File: 2021/go/day7/part1/part1.go
package part1

import (
	"sort"
	"strconv"
	"strings"
)

func Solve(input string) string {
	var points []int
	for _, point := range strings.Split(input, ",") {
		n, _ := strconv.Atoi(point)
		points = append(points, n)
	}
	return strconv.Itoa(calc(points))
}

func calc(points []int) int {
	sort.Ints(points)
	median := points[(len(points)+1)/2]

	sum := 0
	for _, point := range points {
		sum += abs(point - median)
	}
	return sum
}

func abs(n int) int {
	if n < 0 {
		return -n
	}
	return n
}

File: 2021/elixir/apps/day7/lib/part1.ex
defmodule Day7Part1 do
  @moduledoc """
  Day7 - Part1
  """

  require Integer

  def solve(input) do
    input
    |> String.split(",")
    |> Enum.map(&String.to_integer/1)
    |> settle()
  end

  defp settle(points) do
    median = median(points)
    Enum.reduce(points, 0, fn point, acc -> acc + abs(point - median) end)
  end

  defp median(points) do
    points |> Enum.sort() |> Enum.at(ceil(length(points) / 2))
  end
end
